code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -97,32 +97,38 @@
 	if(last_fire && last_fire + 15 SECONDS > world.time)
 		. = span_notice("[user] touches the end of [src] to \the [A], using the residual heat to ignite it in a puff of smoke. What a badass.")
 
-/obj/item/gun/ballistic/revolver/detective
-	name = "\improper Colt Detective Special"
-	desc = "A classic, if not outdated, law enforcement firearm. Uses .38 Special rounds. \nSome spread rumors that if you loosen the barrel with a wrench, you can \"improve\" it."
-	fire_sound = 'sound/weapons/gun/revolver/shot.ogg'
-	icon_state = "detective"
+/obj/item/gun/ballistic/revolver/c38
+	name = "\improper .38 revolver"
+	desc = "A classic, if not outdated, lethal firearm. Uses .38 Special rounds."
 	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev38
 	initial_caliber = CALIBER_38
-	alternative_caliber = CALIBER_357
+	icon_state = "c38"
+	fire_sound = 'sound/weapons/gun/revolver/shot.ogg'
 	initial_fire_sound = 'sound/weapons/gun/revolver/shot.ogg'
```

Comment:
these initial variables should be on the detective subtype. they only matter for weapons with multiple modes